app/controllers/AdminController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+//require_once 'models/PostModel.php';
+//require_once 'models/CommentModel.php';
+//require_once 'models/UserModel.php';
+namespace app\controllers;
+use app\models\CommentModel;
+use app\models\PostModel;
+use app\models\UserModel;
+use app\utils\View;
+
+class AdminController {
+
+    private PostModel $postModel;
+    private CommentModel $commentModel;
+    private UserModel $userModel;
+
+    public function __construct() {
+        $this->postModel = new PostModel();
+        $this->commentModel = new CommentModel();
+        $this->userModel = new UserModel();
+    }
+
+    public function index(): void
+    {
+        $posts = $this->postModel ->getPosts();
+//        require_once 'views/admin/dashboard.php';
+        View::render('views/admin/dashboard.php', ['posts' => $posts]);
+    }
+
+    public function show($id): void
+    {
+        $post = $this->postModel->getPostByID($id);
+        $comments = $this->commentModel->getApprovedComments($id);
+        if ($post == null) {
+            $post = [];
+        }
+        if ($comments == null) {
+            $comments = [];
+        }
+//        var_dump($id);
+//        var_dump($post);
+//        var_dump($comment);
+//        require_once 'views/admin/admin_show.php';
+        View::render('views/admin/admin_show.php', ['post' => $post, 'comments' => $comments]);
+    }
+
+    public function users(): void
+    {
+        $users = $this->userModel->findAllUsers();
+//        require_once 'views/admin/usermanagement.php';
+        View::render('views/admin/usermanagement.php', ['users' => $users]);
+    }
+
+    public function create(): void
+    {
+        if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+            $username = $_POST['username'];
+            $email = $_POST['email'];
+            $password = $_POST['password'];
+
+            if ($this->userModel->findUserByUsername($username)) {
+                echo "Username is already taken.";
+                return;
+            }
+
+            if ($this->userModel->findUserByEmail($email)) {
+                echo "Email is already registered";
+                return;
+            }
+
+            $this->userModel->register($username, $email, $password);
+            header("Location: /admin/users");
+        } else {
+            $this->tosignup();
```

Comment:
@tony-tang-aligent I don't see this function in this class?

Reply:
For this project's setting, I just assume we have a built-in sysadmin user, so the only way to create a new admin is to login an existing admin and modify the role of an existing user to an admin. So there is no way we can hit the else branch. And I simply copy and paste the logic from creating a normal user, so I think I don't need the else clause for this reason. So there is no such function =.=

Reply:
Let's remove it then